lib/checks/keyboard/main-is-top-level.js

```diff
@@ -0,0 +1,13 @@
+var landmarks = axe.commons.aria.getRolesByType('landmark');
+var parent = axe.commons.dom.getComposedParent(node);
+while (parent){
+	var role = parent.getAttribute('role');
+	if (!role && (parent.tagName.toLowerCase() !== 'form')){
```

Comment:
This needs to work with `role=form` too.

Reply:
Does line 5 not accomplish that?